### PR TITLE
ROX-25813: Add compliance coverages page status filtering

### PR DESCRIPTION
**Pull request by bradr5**
### Description

Add the ability to filter by status on the compliance coverages page tables.

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

Without filters:
Screenshot 2024-10-07 at 2 55 59 PM

With 'Pass' filter applied:
Screenshot 2024-10-07 at 2 56 08 PM

With 'Pass' and 'Manual' filter applied:
Screenshot 2024-10-07 at 2 56 21 PM

Compliance header tooltip:
Screenshot 2024-10-07 at 2 56 27 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/StatusCountIcon.tsx [hunk 1]
```diff
@@ -18,11 +18,12 @@ type StatusCountIconProps = {
     text: string;
     status: Status;
     count: number;
+    disabled?: boolean;
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/StatusCountIcon.tsx:21]: A question for your thoughts, not a change request.

What are pro and con of `disabled` as optional? Pardon pun on `?`

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx [hunk 2]
```diff
@@ -145,50 +165,66 @@ function ProfileClustersTable({
...
-                                                            {`${passCount} / ${totalCount} checks are passing for this cluster`}
-                                                        </div>
-                                                    }
-                                                    triggerRef={() =>
-                                                        document.getElementById(
-                                                            progressBarId
-                                                        ) as HTMLButtonElement
-                                                    }
-                                                />
+                                                {isComplianceStatusFiltered() ? (
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx:202]: What are pro and con for future refactoring as a shared element, as you did for `StatusCountIcon`?

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx [hunk 3]
```diff
@@ -48,8 +49,20 @@ function ProfileClustersTable({
...
 }: ProfileClustersTableProps) {
     const { generatePathWithScanConfig } = useScanConfigRouter();
+    const { searchFilter } = useURLSearch();
     const { page, perPage, setPage, setPerPage } = pagination;
 
+    function isComplianceStatusFiltered() {
+        return Array.isArray(searchFilter['Compliance Check Status']);
+    }
+
+    function shouldDisableIcon(statuses) {
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx:59]: Interesting that I did not find a type for these. The strings correspond to `statusText` properties?
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/compliance.coverage.utils.tsx#L118-L201

And I am unfamiliar enough with search filter to know where else to take advantage of a type.

**Review by pedrottimark — APPROVED** [review]
Clear changes. A few comments partly to keep myself alert in late afternoon :)

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/StatusCountIcon.tsx (hunk 1 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/StatusCountIcon.tsx:21]: I thought of `StatusCountIcon` as more of a general reusable compliance component, where the disabled prop might not always be necessary, especially when filtering isn’t involved. I know it’s best not to code for uncertain future needs, but that was my thought process.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx (hunk 2 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx]: I think it's a great idea. Went ahead and created a `ComplianceProgressBar` component.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx (hunk 3 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx:49]: Yeah this bothers me as well. Our backend uses all caps for status labels (e.g., PASS), but when we implemented filtering, we used standard capitalization (e.g., Pass), leading to the inconsistency. I think ideally we should display the status in normal capitalization in the dropdown selection, but convert it to uppercase in the URL parameter..?

**Pull request merged**